spark/src/main/scala/org/apache/spark/sql/delta/Snapshot.scala

```diff
@@ -367,8 +377,12 @@ class Snapshot(
 
   /** The [[CheckpointProvider]] for the underlying checkpoint */
   lazy val checkpointProvider: CheckpointProvider = logSegment.checkpointProvider match {
-    case cp: CheckpointProvider => cp
-    case o => throw new IllegalStateException(s"Unknown checkpoint provider: ${o.getClass.getName}")
+    case cp: CheckpointProvider =>
+      cp
+    case uninitializedProvider: UninitializedCheckpointProvider =>
+      CheckpointProvider(spark, this, checksumOpt, uninitializedProvider)
+    case o =>
+      throw new IllegalStateException(s"Unknown checkpoint provider: ${o.getClass.getName}")
```

Comment:
nit: any particular reason to churn the two existing cases?

Reply:
reverted.